File: Hash/424.Longest-Repeating-Character-Replacement/424.Longest-Repeating-Character-Replacement.py
class Solution:
    def characterReplacement(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: int
        """
        counts=collections.defaultdict(int)
        lo, result, maxn = 0,0,0
        for hi in range(len(s)):
            counts[s[hi]]+=1
            maxn = max(maxn, counts[s[hi]])
            if hi-lo+1-maxn>k:                
                counts[s[lo]]-=1
                lo+=1
            else:
                result = max(result,hi-lo+1)
        return result
        

File: Hash/424.Longest-Repeating-Character-Replacement/424.Longest-Repeating-Character-Replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) 
    {
        vector<int>Count(26,0);        
        int i = 0;        
        int result = 0;
        
        for (int j=0; j<s.size(); j++)
        {
            Count[s[j]-'A']++;
            int majorityNum = *max_element(Count.begin(),Count.end());
            
            if (j-i+1-majorityNum <=k)
                result = max(result, j-i+1);
            else
            {
                Count[s[i]-'A']--;
                i++;
            }
            
        }
        return result;
    }
};
